src/Runner.Worker/ExecutionContext.cs

```diff
@@ -856,7 +856,11 @@ private void InitializeTimelineRecord(Guid timelineId, Guid timelineRecordId, Gu
             var configuration = HostContext.GetService<IConfigurationStore>();
             _record.WorkerName = configuration.GetSettings().AgentName;
 
-            _jobServerQueue.QueueTimelineRecordUpdate(_mainTimelineId, _record);
+            // If we are inside an action, we don't want the steps inside to update the displayname of the parent so we pass the same parent displayname.
```

Comment:
should we not call `InitializeTimelineRecord` when inside an action?

Reply:
@TingluoHuang we need to call it so that the step info is shown while the step is running. 

If you don't call `InitializeTimelineRecord`, the composite action node will not expand with the steps' results while it's running.

Reply:
(resolved in zoom call)